### PR TITLE
feat(retry): 3x initial delay on 429s

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[1667](https://github.com/metriport/metriport-internal/issues/1667)

### Description

- longer initial delay on 429s 

### Testing

- Local
  - [x] added unit tests

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/shared/src/net/__tests__/retry.test.ts [hunk 1]
```diff
@@ -187,4 +188,38 @@ describe("executeWithNetworkRetries", () => {
...
+      error.response = {
+        status: 429,
+        data: {},
+        statusText: "Too Many Requests",
+        headers: {},
+        config: {
+          headers: new AxiosHeaders(),
+          method: "post",
+          url: "",
+        },
```
Comment (jonahkaye) [inline packages/shared/src/net/__tests__/retry.test.ts:206]: I needed to pass all this stuff in to conform to Axios typing

File: packages/shared/src/net/__tests__/retry.test.ts [hunk 2]
```diff
@@ -187,4 +188,38 @@ describe("executeWithNetworkRetries", () => {
...
+      executeWithNetworkRetries(fn, {
+        initialDelay: 1,
+        maxAttempts: 2,
+      })
+    ).rejects.toThrow();
+
+    expect(fn).toHaveBeenCalledTimes(2);
+    expect(spyDefaultGetTimeToWait).toHaveBeenCalledWith(
+      expect.objectContaining({
+        initialDelay: 3,
```
Comment (jonahkaye) [inline packages/shared/src/net/__tests__/retry.test.ts:221]: ensuring delay was 3x as long

File: packages/shared/src/net/retry.ts [hunk 3]
```diff
@@ -25,10 +30,52 @@ const defaultOptions: ExecuteWithNetworkRetriesOptions = {
     "ECONNREFUSED", // (Connection refused): No connection could be made because the target machine actively refused it. This usually results from trying to connect to a service that is inactive on the foreign host.
     "ECONNRESET", //  (Connection reset by peer): A connection was forcibly closed by a peer. This normally results from a loss of the connection on the remote socket due to a timeout or reboot. Commonly encountered via the http and net modules.
   ],
-  httpStatusCodesToRetry: [429], // 429 Too Many Requests
+  httpStatusCodesToRetry: [tooManyRequestsStatus], // 429 Too Many Requests
```
Comment (thomasyopes) [inline packages/shared/src/net/retry.ts:33]: You can remove the comment

File: packages/shared/src/net/retry.ts [hunk 4]
```diff
@@ -25,10 +30,52 @@ const defaultOptions: ExecuteWithNetworkRetriesOptions = {
...
+ * @param maxDelay The maximum delay in milliseconds.
+ * @param error The error that occurred.
+ * @returns The time to wait before the next retry.
+ */
+function networkGetTimeToWait({
+  initialDelay,
+  backoffMultiplier,
+  attempt,
+  maxDelay,
+  error,
```
Comment (thomasyopes) [inline packages/shared/src/net/retry.ts:59]: What are you doing with this error?

File: packages/shared/src/common/retry.ts [hunk 5]
```diff
@@ -107,7 +108,13 @@ export async function executeWithRetries<T>(
         throw error;
       }
       log(`${msg}, retrying... (attempt: ${attempt})`);
-      const timeToWait = getTimeToWait({ initialDelay, backoffMultiplier, attempt, maxDelay });
+      const timeToWait = getTimeToWait({
+        initialDelay,
+        backoffMultiplier,
+        attempt,
+        maxDelay,
+        error,
```
Comment (thomasyopes) [inline packages/shared/src/common/retry.ts:116]: Same comment as below -- adding error but not seeming to change any code to do anything with it.

File: packages/shared/src/net/retry.ts (hunk 4 above)
Comment (jonahkaye) [reply packages/shared/src/net/retry.ts:59]: we are choosing a different delay based on the error ...

File: packages/shared/src/common/retry.ts (hunk 5 above)
Comment (jonahkaye) [reply packages/shared/src/common/retry.ts:116]: responded there but the initialDelay is dependent on the error

File: packages/shared/src/net/retry.ts [hunk 6]
```diff
@@ -25,10 +30,45 @@ const defaultOptions: ExecuteWithNetworkRetriesOptions = {
...
   retryOnTimeout: false,
 };
 
+function getHttpStatusFromError(error: unknown): number | undefined {
+  return axios.isAxiosError(error) ? error.response?.status : undefined;
+}
+
+function getHttpCodeFromError(error: unknown): string | undefined {
+  return axios.isAxiosError(error) ? error.code : undefined;
+}
```
Comment (leite08) [inline packages/shared/src/net/retry.ts:43]: 😍

**Review by leite08 — APPROVED** [review]
🚀

**Pull request merged**